ndcube/ndcube.py

```diff
@@ -1,40 +1,131 @@
+
+        origin : `int`.
+            Origin of the top-left corner. i.e. count from 0 or 1.
+            Normally, origin should be 0 when passing numpy indices, or 1 if
+            passing values from FITS header or map attributes.
+            See `~astropy.wcs.WCS.wcs_pix2world` for more information.
+            Default is 0.
+
+        Returns
+        -------
+
+        coord : `list`
+            A list of arrays containing the output coordinates
+            reverse of the wcs axis order.
+        """
+
+    @abc.abstractmethod
+    def world_to_pixel(self, quantity_axis_list, origin=0):
+        """
+        Convert a world coordinate to a data (pixel) coordinate by using
+        `~astropy.wcs.WCS.all_world2pix`.
+
+        Parameters
+        ----------
+        quantity_axis_list : `list`
+            A list of `~astropy.units.Quantity`.
+
+        origin : `int`
+            Origin of the top-left corner. i.e. count from 0 or 1.
+            Normally, origin should be 0 when passing numpy indices, or 1 if
+            passing values from FITS header or map attributes.
+            See `~astropy.wcs.WCS.wcs_world2pix` for more information.
+            Default is 0.
+
+        Returns
+        -------
+
+        coord : `list`
+            A list of arrays containing the output coordinates
+            reverse of the wcs axis order.
+        """
+
+    @abc.abstractproperty
+    def dimensions(self):
+        """
+        Returns a named tuple with two attributes: 'shape' gives the shape
+        of the data dimensions; 'axis_types' gives the WCS axis type of each dimension,
+        e.g. WAVE or HPLT-TAN for wavelength of helioprojected latitude.
+        """
 
-DimensionPair = namedtuple('DimensionPair', 'shape axis_types')
-SequenceDimensionPair = namedtuple('SequenceDimensionPair', 'shape axis_types')
+    @abc.abstractmethod
+    def crop_by_coords(self, lower_left_corner, dimension_widths):
+        """
+        Crops an NDCube given a lower left corner and widths of region of interest.
+
+        Parameters
+        ----------
+        lower_left_corner: `list` of `astropy.units.Quantity`
+            The lower left corner of the region of interest described in physical units
+            consistent with the NDCube's wcs object.  The length of the iterable must
+            equal the number of data dimensions and must have the same order as the data.
 
-__all__ = ['NDCube', 'NDCubeSequence', 'SequenceDimensionPair', '_extra_coords_to_input_format']
+        dimension_widths: iterable of `astropy.units.Quantity`
+            The width of the region of interest in each dimension in physical units
+            consistent with the NDCube's wcs object.  The length of the iterable must
+            equal the number of data dimensions and must have the same order as the data.
 
+        Returns
+        -------
+        result: NDCube
 
-class NDCube(astropy.nddata.NDData):
+        """
+
+
+class NDCube(NDCubeBase):
```

Comment:
I think it might be a good idea to document somewhere in the ```NDCube``` attributes that the docstrings can be found in the ```NDCubeBase``` class so people ca easily find where to look when searching the codebase.  What do you think?